introduction-to-dom-and-its-manipulation.html

```diff
@@ -0,0 +1,358 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="UTF-8">
+<title>Introduction to DOM and its Manipulation - iodide</title>
+<link rel="stylesheet" type="text/css" href="https://iodide-project.github.io/master/iodide.master.css">
+</head>
+<body>
+<script id="jsmd" type="text/jsmd">
+%% meta
+{
+  "title": "Introduction to DOM and its Manipulation",
+  "lastSaved": "2018-03-07T21:10:40.042Z",
+  "lastExport": "2018-03-07T21:16:36.346Z"
+}
+
+%% md
+# Introduction to DOM and its Manipulation
+
+%% md
+## What is DOM ?
+
+The backbone of an HTML document is tags.  
+According to Document Object Model (DOM), every HTML-tag is an object. These objects then can be manipulated using a scripting language such as JavaScript.
+
+%% md
+## Why should I care about DOM anyway ? 
+
+DOM comes with a set of associated APIs through which its properties can be manipulated and configured as required. This enables us to make DOM structures (such as webpages or this notebook) interactive. Using Iodide a user can work directly on the DOM which was not quite possible using other traditional notebooks.
+
+This notebook briefly describes how one can do manipulations on the DOM using Iodide.  
+We will try to learn and understand the workings of DOM by example. First let us make a block using the **div** tag
+
+%% md
+<div id="test">
+  Here we have defined a div tag with an id of test
+</div>
+
+This creates a new node in DOM tree which can be uniquely identified using the **id** *test*.
+
+Lets add some CSS to make this look better
+
+%% css
+#test {
+  padding: 1em;
+  margin: 1em;
+  width: 100px;
+  background: #bdc3c7
+}
+
+%% md
+We will now perform a very simple manipulation to the above created element using JavaScript
+
+%% js
+// Get element from the DOM tree
+a = document.getElementById('test')
+// Change a style property
+a.style.background = "#3498db"
+
+%% md
+As you would have noticed, the background color of the box has changed without the need of re-rendering that cell. DOM APIs consits of functions similar to `getElementById` such as `querySelectorAll`, `querySelector`, `getElementsByTagName` etc. which provides us access to different elements on the DOM tree.
+
+How about we make something more interactive.  
```

Comment:
"How about we make something more interactive." is a question:
"How about we make something more interactive?"